core/src/database/SQLite3Backend.cpp

```diff
@@ -69,13 +68,13 @@ namespace ledger {
                 throw make_exception(api::ErrorCode::DATABASE_EXCEPTION, "Database should be initiated before changing password.");
             }
 
-            std::string db_params;
+            auto db_params = fmt::format("dbname=\"{}\" ", _dbResolvedPath) + fmt::format("key=\"{}\" ", oldPassword);
             if (!newPassword.empty()) {
                 // change password if a new password is provided
-                db_params = fmt::format("dbname=\"{}\" ", _dbResolvedPath) + fmt::format("key=\"{}\" ", oldPassword) + fmt::format("new_key=\"{}\" ", newPassword);
+                db_params += fmt::format("new_key=\"{}\" ", newPassword);
```

Comment:
Just a small comment: your change will make that string to allocate twice while the previous code allocated only once. While doing that?

Reply:
+1, but here I did this to clearly see/be sure that we use same (base) params plus their differences, to me it's an "optim" that we can make once the feature is bug free